cloud/azure/azure_rm_networkinterface.py

```diff
@@ -0,0 +1,619 @@
+            public_ip=dict(type='bool', default=True),
+            subnet_name=dict(type='str', aliases=['subnet']),
+            virtual_network_name=dict(type='str', aliases=['virtual_network']),
+            os_type=dict(type='str', choices=['Windows', 'Linux'], default='Linux'),
+            open_ports=dict(type='list'),
+            public_ip_allocation_method=dict(type='str', choices=['Dynamic', 'Static'], default='Dynamic'),
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.location = None
+        self.security_group_name = None
+        self.private_ip_address = None
+        self.private_ip_allocation_method = None
+        self.public_ip_address_name = None
+        self.state = None
+        self.subnet_name = None
+        self.tags = None
+        self.virtual_network_name = None
+        self.security_group_name = None
+        self.os_type = None
+        self.open_ports = None
+        self.public_ip_allocation_method = None
+        self.public_ip = None
+
+        self.results = dict(
+            changed=False,
+            results=dict(),
+        )
+
+        super(AzureRMNetworkInterface, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                      supports_check_mode=True)
+
+    def exec_module(self, **kwargs):
+
+        for key in self.module_arg_spec.keys() + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        self.results['check_mode'] = self.check_mode
+
+        results = dict()
+        changed = False
+        nic = None
+        subnet = None
+        nsg = None
+        pip = None
+
+        resource_group = self.get_resource_group(self.resource_group)
+        if not self.location:
+            # Set default location
+            self.location = resource_group.location
+
+        if not NAME_PATTERN.match(self.name):
+            self.fail("Parameter error: name must begin with a letter or number, end with a letter or number "
+                      "and contain at least one number.")
+
+        if self.state == 'present':
+            if self.virtual_network_name and not self.subnet_name:
+                self.fail("Parameter error: a subnet is required when passing a virtual_network_name.")
+
+            if self.subnet_name and not self.virtual_network_name:
+                self.fail("Parameter error: virtual_network_name is required when passing a subnet value.")
+
+            if self.virtual_network_name and self.subnet_name:
+                subnet = self.get_subnet(self.virtual_network_name, self.subnet_name)
+
+            if self.public_ip_address_name:
+                pip = self.get_public_ip_address(self.public_ip_address_name)
+
+            if self.security_group_name:
+                nsg = self.get_security_group(self.security_group_name)
+
+        try:
+            self.log('Fetching network interface {0}'.format(self.name))
```

Comment:
format is not 2.4 friendly, but I'm going to assume that azure library already requires +2.6 so we can ignore 2.4 compatibility.